substrate/frame/preimage/src/benchmarking.rs

```diff
@@ -43,206 +42,225 @@ fn sized_preimage_and_hash<T: Config>(size: u32) -> (Vec<u8>, T::Hash) {
-	}: note_preimage(RawOrigin::Signed(caller), preimage)
-	verify {
-		assert!(Preimage::<T>::have_preimage(&hash));
+
+		#[extrinsic_call]
+		note_preimage(RawOrigin::Signed(caller), preimage);
+
+		assert!(Pallet::<T>::have_preimage(&hash));
 	}
+
 	// Cheap note - will not reserve since it's the manager.
-	note_no_deposit_preimage {
-		let s in 0 .. MAX_SIZE;
+	#[benchmark]
+	fn note_no_deposit_preimage(s: Linear<0, MAX_SIZE>) {
+		let o = T::ManagerOrigin::try_successful_origin()
+			.expect("ManagerOrigin has no successful origin required for the benchmark");
 		let (preimage, hash) = sized_preimage_and_hash::<T>(s);
-		assert_ok!(Preimage::<T>::request_preimage(
-			T::ManagerOrigin::try_successful_origin()
-				.expect("ManagerOrigin has no successful origin required for the benchmark"),
-			hash,
-		));
-	}: note_preimage<T::RuntimeOrigin>(
-		T::ManagerOrigin::try_successful_origin().map_err(|_| BenchmarkError::Weightless)?,
-		preimage
-	) verify {
-		assert!(Preimage::<T>::have_preimage(&hash));
+		assert_ok!(Pallet::<T>::request_preimage(o.clone(), hash,));
+
+		#[extrinsic_call]
+		note_preimage(o as T::RuntimeOrigin, preimage);
+
+		assert!(Pallet::<T>::have_preimage(&hash));
 	}
 
 	// Expensive unnote - will unreserve.
-	unnote_preimage {
+	#[benchmark]
+	fn unnote_preimage() {
 		let caller = funded_account::<T>();
 		let (preimage, hash) = preimage_and_hash::<T>();
-		assert_ok!(Preimage::<T>::note_preimage(RawOrigin::Signed(caller.clone()).into(), preimage));
-	}: _(RawOrigin::Signed(caller), hash)
-	verify {
-		assert!(!Preimage::<T>::have_preimage(&hash));
+		assert_ok!(Pallet::<T>::note_preimage(RawOrigin::Signed(caller.clone()).into(), preimage));
+
+		#[extrinsic_call]
+		_(RawOrigin::Signed(caller), hash);
+
+		assert!(!Pallet::<T>::have_preimage(&hash));
 	}
+
 	// Cheap unnote - will not unreserve since there's no deposit held.
-	unnote_no_deposit_preimage {
+	#[benchmark]
+	fn unnote_no_deposit_preimage() {
+		let o = T::ManagerOrigin::try_successful_origin()
+			.expect("ManagerOrigin has no successful origin required for the benchmark");
 		let (preimage, hash) = preimage_and_hash::<T>();
-		assert_ok!(Preimage::<T>::note_preimage(
-			T::ManagerOrigin::try_successful_origin()
-				.expect("ManagerOrigin has no successful origin required for the benchmark"),
-			preimage,
-		));
-	}: unnote_preimage<T::RuntimeOrigin>(
-		T::ManagerOrigin::try_successful_origin().map_err(|_| BenchmarkError::Weightless)?,
-		hash
-	) verify {
-		assert!(!Preimage::<T>::have_preimage(&hash));
+		assert_ok!(Pallet::<T>::note_preimage(o.clone(), preimage,));
+
+		#[extrinsic_call]
+		unnote_preimage(o as T::RuntimeOrigin, hash);
+
+		assert!(!Pallet::<T>::have_preimage(&hash));
 	}
 
 	// Expensive request - will unreserve the noter's deposit.
-	request_preimage {
+	#[benchmark]
+	fn request_preimage() {
+		let o = T::ManagerOrigin::try_successful_origin()
+			.expect("ManagerOrigin has no successful origin required for the benchmark");
```

Comment:
Why is this an expect now? The pallet is also usable without manager origin.  
It returned `BenchmarkError::Weightless` before.

Reply:
I think that error is meaningless.

Same as the https://github.com/paritytech/polkadot-sdk/blob/06debd0b408cc4308079d9f980245939697fdfa8/substrate/frame/treasury/src/benchmarking.rs#L121-L126.

It was not part of the benchmark code block. Additionally, I believe the original `.expect` message is more readable. So I reuse it.

What do you suggest?

Reply:
Also can be found here

https://github.com/paritytech/polkadot-sdk/blob/06debd0b408cc4308079d9f980245939697fdfa8/substrate/frame/lottery/src/benchmarking.rs#L50